### ISSUE TITLE
quit(keep_kernel=True) doesn't work

### ISSUE THREAD

**Issue by zer0n** [ipython/ipython#8482]
I'm using IPython 3.1.0

**Comment by takluyver** [comment]
Which interface are you doing this in? In what way does it not work? Details are useful!

**Comment by zer0n** [comment]
I was working on a notebook and needed to debug so I ran `ipython console --existing kernel-ad2bfcc4-5246-43f7-9ddf-3189506b23d8.json`.

Then I tried `quit(keep_kernel=True)` from the console IPython and the notebook kernel got restarted (everytime).

**Comment by Carreau** [comment]
Can reproduce with the console also:
```
~/d/ipython (3.x) $ ipython console
IPython Console 3.2.0-dev

In [1]: quit?
Type:           ZMQExitAutocall
String form:    <IPython.core.autocall.ZMQExitAutocall object at 0x109fd0278>
File:           ~/dev/ipython/IPython/core/autocall.py
Signature:      quit(self, keep_kernel=False)
Docstring:
Exit IPython. Autocallable, so it needn't be explicitly called.

Parameters
----------
keep_kernel : bool
  If True, leave the kernel alive. Otherwise, tell the kernel to exit too
  (default).
Call signature: quit(keep_kernel=False)

In [3]: quit(keep_kernel=True)
~/d/ipython (3.x) $ ipython console --existing
[ZMQTerminalIPythonApp] CRITICAL | Could not find existing kernel connection file kernel*.json
```

Quit is an alias for exit, you can `/quit/exit/` 

From a quick glance at the code the handeling of keepkernel has never been implemented,

**Comment by Carreau** [comment]
Want to try  #8483 ?

**Comment by takluyver** [comment]
Fixed by #8483
